---
output:
  md_document:
    variant: markdown_github
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "README-"
)
```

# archaeopteryx

The goal of archaeopteryx is to wrap the wonderful [archaeopteryx.js](https://sites.google.com/site/cmzmasek/home/software/archaeopteryx-js) library and make it accessible as an htmlwidget. Currently the basics work but I will be exploring changes to the API to try to make it work with R's pipe-based workflow.

## Mini Roadmap

A few thoughts on how to improve this code (no guarantees):

- To be able to get all of the control-elements, we will need to specify the DOM elements.
- If we want to have multiple trees on an RMD page, the DOM elements need to be labelled in the RMB code using htmlwidget tags.
- To do this is not-trivial since the code names are hardcoded in the archeompertyx.js code and at first look they do not seem to be replaces usign the `options` or `settings` features that allow you to customize the chart.
- provide access to some reasonably complete set of options that can be passed to users.
- right now newick strings are passed/parsed. Can a direct JSON blob be transferred in order to save the encode/decode step? THis would also allow us to adorn information on each node.
- be able to specify per-node attributes - which measn being able to pass in the information and using an API to modify the nodes.


## Simple Usage.


```{r, eval=FALSE}
## basic example code
library(archaeopteryx)

tree       <- parse_newick("(mammal,(turtle,rayfinfish,(frog,salamander)))")
treestring <- tree_to_XML(tree)
archaeopteryx(treestring)

```

## phylo XML handling

```{r, eval=FALSE}
write(x = "turtle	TAXONOMY_CODE:BACTN	TAXONOMY_ID:226186	TAXONOMY_ID_PROVIDER:ncbi	TAXONOMY_SN:Bacteroides thetaiotaomicron	SEQ_ACCESSION:29341016	SEQ_ACCESSION_SOURCE:gi SEQ_NAME:SusD SEQ_SYMBOL:BT3701",
      file="temp.txt")

tree  <- parse_newick("(mammal,(turtle,rayfinfish,(frog,salamander)))")
table <- parse_mapping_file("temp.txt")
decorate_phylogeny(tree, table)
treestring <- tree_to_XML(tree)

archaeopteryx(treestring=treestring)
```

## Adorn Trees with Properties

```{r}
print(tree_to_XML(tree))
add_node_property(tree, "turtle", "size", "300")
print(tree_to_XML(tree))

```

